### PR TITLE
Fix invalid JSON during morango sync

### PR DESCRIPTION
**Pull request by bjester**
### Summary

Updates the content negotiation class to use default behavior for morango.

### Reviewer guidance

- Needs a GZIP enabled peer
- Pull a facility first, make changes, then try to sync with the peer you pulled from (it should push)

### References

…

----

### Contributor Checklist

PR process:

- [X] PR has the correct target branch and milestone
- [X] PR has 'needs review' or 'work-in-progress' label
- [X] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [X] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/negotiation.py [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
         """
+
+        if request.path.startswith("/api/morango"):
+            return super(LimitContentNegotiation, self).select_parser(request, parsers)
         if "multipart" in request.content_type:
             return MultiPartParser()
         return JSONParser()
 
-    def select_renderer(self, request, renderers, format_suffix):
+    def select_renderer(self, request, renderers, format_suffix=None):
```
Comment (rtibbles) [inline kolibri/core/negotiation.py:6]: Does it need the renderer? Because it's gzipping the output too?

File: kolibri/core/negotiation.py [hunk 2]
```diff
@@ -1,21 +1,29 @@
...
 
 
-class LimitContentNegotiation(BaseContentNegotiation):
+class LimitContentNegotiation(DefaultContentNegotiation):
     def select_parser(self, request, parsers):
         """
         Always return JSONParser unless a 'multipart/form-data' content type is sent
+        or this is for morango
         """
+
```
Comment (rtibbles) [inline kolibri/core/negotiation.py]: Note, I think we can just remove the `select_parser` override here, and just default to the `DefaultContentNegotiation` parser strategy - it's only the renderer that really needs to be overridden for performance reasons.

Comment (bjester) [reply kolibri/core/negotiation.py]: I can make this change

File: kolibri/core/negotiation.py (hunk 1 above)
Comment (bjester) [reply kolibri/core/negotiation.py:6]: Hmm, I don't think it's needed.

**Comment by bjester** [comment]
@jredrejo Could you verify that removing the `select_parser` override works as expected with whatever prompted your multipart additions?

File: kolibri/core/negotiation.py (hunk 1 above)
Comment (rtibbles) [reply kolibri/core/negotiation.py:6]: So, do we need the special case for morango in the renderer, or just need the parsing change above?

File: kolibri/core/negotiation.py [hunk 3]
```diff
@@ -1,21 +1,15 @@
...
-        if "multipart" in request.content_type:
-            return MultiPartParser()
-        return JSONParser()
-
-    def select_renderer(self, request, renderers, format_suffix):
-        """
-        Always return JSONRenderer
+        Always return JSONRenderer unless for morango
         """
+        if request.path.startswith("/api/morango"):
```
Comment (jamalex) [inline kolibri/core/negotiation.py]: Is there a more robust way to get this URL? Maybe using a `reverse` of some sort? There's a chance Kolibri has been mounted at a different subpath (@rtibbles knows about that process) -- though I'm not sure whether Morango actually supports being mounted like that (whether peers will then discover it correctly, etc).

Comment (bjester) [reply kolibri/core/negotiation.py]: Ah, I didn't know that was a known use case. Okay, I think we could just remove this. I was getting another JSON decode error which I can no longer reproduce.

**Review by rtibbles — APPROVED** [review]

**Pull request merged**